bbc_sale/data/action_automated_emails_magento.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data noupdate="0">
+        <!-- SALES ORDER CONFIRMATION -->
+
+        <!-- automatically sends email after confirming sales order from Magento -->
+        <record id="action_send_email_confirmation_sales_order_magento" model="ir.actions.server">
+            <field name="name">Magento | Send Email After Confirming Sales Order</field>
+            <field name="model_id" ref="sale.model_sale_order"/>
+            <field name="condition">object.create_uid.name == 'magento'</field>
+            <field name="type">ir.actions.server</field>
+            <field name="state">email</field>
+        </record>
+
+        <!-- add action_send_email_confirmation_sales_order_magento to workflow activity confirming sales order -->
+        <record id="sale.act_router" model="workflow.activity"> 
+            <field name="action_id" ref="action_send_email_confirmation_sales_order_magento"/>
+        </record>
+
+        <!-- SALES ORDER CANCELLATION -->
+
+        <!-- automatically sends email after cancellation sales order from Magento -->
+        <record id="action_send_email_cancellation_sales_order_magento" model="ir.actions.server">
+            <field name="name">Magento | Send Email After Cancellation Sales Order</field>
+            <field name="model_id" ref="sale.model_sale_order"/>
+            <field name="condition">object.create_uid.name == 'magento'</field>
+            <field name="type">ir.actions.server</field>
+            <field name="state">email</field>
+        </record>
+
+        <record id="filter_magento_order_canceled" model="ir.filters">
+            <field name="name">Magento | Canceled Sales Orders</field>
+            <field name="model_id">sale.order</field>
+            <field name="domain">[('state', '=', 'cancel'), ('write_uid', 'ilike', 'magento')]</field>
+            <field name="user_id" eval="False" />
+        </record>
+
+        <record id="rule_send_email_cancellation_sales_order_magento" model="base.action.rule">
+            <field name="name">Magento | Send Email After Cancellation Sales Order</field>
+            <field name="model_id" ref="sale.model_sale_order" />
+            <field name="kind">on_create_or_write</field>
+            <field name="filter_id" ref="filter_magento_order_canceled" />
```

Comment:
I think you need a filter_pre_id on state != cancel otherwise the email is sent whenever a write is performed on long cancelled orders too.

Reply:
@StefanRijnhart You are right! I solved this in my latest commit.